File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    n=len(answers)
    
    student_1=[]
    student_2=[]
    student_3=[]
    
    #1번 학생
    for i in range(n):
        if i%5==0:
            student_1.append(1)
        elif i%5==1:
            student_1.append(2)
        elif i%5==2:
            student_1.append(3)
        elif i%5==3:
            student_1.append(4)
        elif i%5==4:
            student_1.append(5)
    
    #2번 학생
    for i in range(n):
        #짝수면 다 2
        if i%2==0:
            student_2.append(2)
        elif i%8==1:
            student_2.append(1)
        elif i%8==3:
            student_2.append(3)
        elif i%8==5:
            student_2.append(4)
        elif i%8==7:
            student_2.append(5)
    
    #3번 학생
    for i in range(n):
        if i%10==0 or i%10==1:
            student_3.append(3)
        elif i%10==2 or i%10==3:
            student_3.append(1)
        elif i%10==4 or i%10==5:
            student_3.append(2)
        elif i%10==6 or i%10==7:
            student_3.append(4)
        elif i%10==8 or i%10==9:
            student_3.append(5)
    
    
    cnt1,cnt2,cnt3=0,0,0
    
    #답 체크하기
    for i in range(n):
        if answers[i]==student_1[i]:
            cnt1+=1
        
        if answers[i]==student_2[i]:
            cnt2+=1
        
        if answers[i]==student_3[i]:
            cnt3+=1
    
    
    #최고 점수 게산
    max_score=max(cnt1,cnt2,cnt3)
    result=[]
    
    #최고 점수를 가진 학생 찾기
    if cnt1==max_score:
        result.append(1)
    if cnt2==max_score:
        result.append(2)
    if cnt3==max_score:
        result.append(3)
    
    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.ArrayList;
import java.util.List;
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] one={1,2,3,4,5};  //5
        int[] two={2,1,2,3,2,4,2,5}; //8
        int[] three={3,3,1,1,2,2,4,4,5,5}; //10
        
        int one_cnt=0;
        int two_cnt=0;
        int three_cnt=0;
        
        //1.점수 게산
        for(int i=0;i<answers.length;i++){
            if(answers[i]==one[i%5]) one_cnt++; //1번학생
            if(answers[i]==two[i%8]) two_cnt++; //2번학생 
            if(answers[i]==three[i%10]) three_cnt++; //3번학생
        }
        
        //2.최댓값 찾기
        //세 점수 중 가장 큰 값을 구한다.
        int maxScore=Math.max(one_cnt, Math.max(two_cnt,three_cnt));
        
        //3.가장 높은 점수를 받은 사람 찾기
        //문제 조건에서 결과는 오름차순이어야 하므로 1번학생부터 차례대로 확인한다.
        List<Integer> winners=new ArrayList<>();
        if(one_cnt==maxScore) winners.add(1);
        if(two_cnt==maxScore) winners.add(2);
        if(three_cnt==maxScore) winners.add(3);
        
        //4.List를 int[] 배열로 변환
        //스트림을 사용하면 한 줄로 변환이 가능하다
        return winners.stream().mapToInt(Integer::intValue).toArray();
    }
}
